torch/onnx/symbolic_opset9.py

```diff
@@ -2701,22 +2701,39 @@ def native_layer_norm(
 
     two_cst = symbolic_helper._generate_wrapped_number(g, 2.0)
     eps_cst = symbolic_helper._generate_wrapped_number(g, eps)
+    eps_dtype = _type_utils.JitScalarType.from_value(eps_cst)
 
     mean = g.op("ReduceMean", input, axes_i=axes)
     numerator = sub(g, input, mean)
+    # Cast it to eps dtype to avoid precision loss
+    cast_numerator = g.op(
```

Comment:
Would it be necessary to cast all the time, or only when numerator is float16?

Reply:
In torch, EPS is always Double, while in ONNX, we used to keep it Float32 all the time without any issue. This Cast would make us stay with Double all the time.

Reply:
Since EPS is scalar I suspect torch would down cast it when computing, if it is following the same casting logic for arithmetic regarding python number and torch.tensor. You might want to check out `aten/src/ATen/native/cpu/layer_norm_kernel.cpp`

Reply:
Done